### DISCUSSION TITLE
[BUG] Swagger documents cannot be accessed directly through the browser after setting the proxy using root_path

### DISCUSSION THREAD

**Discussion by insistence** [fastapi/fastapi#11626] [Questions] [answered]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Operating System
Windows

### FastAPI Version
0.109.1

### Pydantic Version
2.5.3

### Python Version
3.9.18

### Uvicorn Version
0.27.0
```
from fastapi import FastAPI
import uvicorn


app = FastAPI()


@app.get('/test')
def test():
    return {'test': 'test content'}


if __name__ == '__main__':
    uvicorn.run(app='app:app', root_path='/proxy')
```
When I run the above code, I am unable to access the Swagger document through http://127.0.0.1:8000/docs or http://127.0.0.1:8000/proxy/docs in my browser(Not Found /proxy/openapi.json).

**Comment by chyok** [comment]
Hi, this is the expected result. You can check the documentation here, which explains in detail why it is this way.

https://fastapi.tiangolo.com/advanced/behind-a-proxy/#check-the-docs-ui

**Comment by insistence** [comment]
> Hi, this is the expected result. You can check the documentation here, which explains in detail why it is this way.
> 
> https://fastapi.tiangolo.com/advanced/behind-a-proxy/#check-the-docs-ui

Ah, I know what you mentioned. The current issue is that using a proxy prefix is also inaccessible.

**Comment by YuriiMotov** [comment, marked as answer]
In short, this is designed in such way as to make it working behind the stripping path proxy.
If you run it behind the proxy as it's described in docs, it will work.
But if you try to access it directly, Swagger is trying to find `openapi.json` at `/proxy/openapi.json` (FastAPI knows about `root_path` and adds prefix to all URLs). Uvicorn adds `/proxy` prefix one more time (now it's `/proxy/proxy/openapi.json`) and this causes the problem.

When you run your app behind the stripping path proxy, Swagger will send request to `/proxy/openapi.json`, proxy will strip the prefix and pass `/openapi.json` to Uvicorn. Uvicorn will add that prefix again and the final path will be `/proxy/openapi.json` which is Ok.
